client/components/profilePage/allFriendsModal.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from 'react';
+import { Modal, Image, Button, Table, Badge, Spinner, Row, Col } from 'react-bootstrap';
+import axios from 'axios';
+
+export default function AllFriends({
+  friendsList,
+  handleClose,
+  show,
+}) {
+
+
+
+  return friendsList === undefined ? (
```

Comment:
No need for spinner here if we are not fetching data from the database. All this is synchronous code, so it should take milliseconds. The spinner will pretty much not exist for the user. If you are planning on implementing some network request to get data, then spinner can stay. 

Add ticket to backlog to refactor and we can deal with it later!